customer-service/src/main/scala/domain/services/CustomerService.scala

```diff
@@ -4,17 +4,37 @@ import cats.MonadThrow
 import cats.data.OptionT
 import cats.syntax.all._
 import domain.repositories.CustomerRepository.CustomerRepository
-import domain.{Customer, CustomerId, CustomerWithDetails}
+import domain.{Customer, CustomerDetails, CustomerId, CustomerWithDetails}
 
 class CustomerService[F[_]: MonadThrow](repository: CustomerRepository[F])(
-    customerDetailsService: CustomerDetailsService[F]
+    customerDetailsServices: CustomerDetailsService[F]*
```

Comment:
yes, I put a trap here :)

let me show u analogy to real life, suppose that u r buying apple watch, iphone and airpod max in an offer aggregator like http://galaxus.ch

after u pay the order u get info from galaxus:
1. for iphone u need to go to digitec store in zurich
2. for apple watch u need to go to geneva apple store
3. for airpod max u need to go small reseller in Bazel

for u as a **consumer** of something it is not important in what places are products and how to get them; u asked galaxus about it, so it should received them and u can pick-up all of them at once at for example Lausanne galaxus store

I don't know if analogy was clear, u should have only one `CustomerDetailsService` here, and all appropriate logic should be in `CustomerDetailsService`

`CustomerService` has 0 interest in scatter and gather logic, has 1 interest - to get details

is that clear?